### PR TITLE
Update ZenFiles to New Version 

### PR DESCRIPTION
**Pull request by ayush714**
Update ZenFiles to New / Latest ZenML Version

### CODE REVIEW

File: customer-churn/kubeflow_requirements.txt [hunk 1]
```diff
@@ -1,9 +1,8 @@
 feature_engine==1.2.0
```
Comment (stefannica) [inline customer-churn/kubeflow_requirements.txt:1]: There is no difference I can find between this file and the `requirements.txt` file. Can you maybe remove this, so we only have one ? Make sure to update the `seldon_deployment_pipeline.py` and `training_pipelines.py` files to point to `requirements.txt` instead.

File: customer-churn/kubeflow_requirements.txt [hunk 2]
```diff
@@ -1,9 +1,8 @@
...
-statsmodels
 xgboost
-zenml==0.7.3
+zenml==0.8.1rc0
+optuna   
 sklearn 
-optuna  
\ No newline at end of file
+facets_overview
+mlflow
```
[the remark this answers is not in the document]
Comment (stefannica) [reply customer-churn/kubeflow_requirements.txt:8]: By the way, I don't think mlflow is even actively used in the customer-churn zenfiles. I would remove it altogether if not required.

File: customer-churn/run_seldon_deployment_pipeline.py [hunk 3]
```diff
@@ -119,7 +114,7 @@ def main(
                     model_name=model_name,
                 )
             ),
-            predictor=predictor().with_return_materializers({"predictions": cs_materializer}),
+            predictor=predictor(),  # .with_return_materializers({"predictions": cs_materializer}),
```
Comment (stefannica) [inline customer-churn/run_seldon_deployment_pipeline.py:117]: You should remove the commented out code if no longer needed

**Review by stefannica — CHANGES_REQUESTED** [review]
I found that the requirements handling could still be improved to make it less confusing.

There is also some outdated information in the customer-churn README.md file. I found that if the user tries to follow those instructions to run the example, they don't work:

* some broken/empty links
* the Seldon Core stack setup is missing the ZenML secret configuration (step 2.)
* port-forwarding for the Kubeflow UI doesn't have to be manually configured (step 5). A `zenml stack up` is sufficient.

File: customer-churn/kubeflow_requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ayush714) [reply customer-churn/kubeflow_requirements.txt:8]: Yeah, it make sense, the reason why I added like this was when I was running kubeflow pipeline, I think i was getting module not found error, that's why I added it like this.

File: customer-satisfaction/requirements.txt [hunk 4]
```diff
@@ -4,4 +4,4 @@ lightgbm==3.3.2
 optuna==2.10.0
 streamlit==1.8.1
 xgboost==1.5.2
-zenml==0.7.3
\ No newline at end of file
+zenml==0.8.0
```
[the remark this answers is not in the document]
Comment (stefannica) [reply customer-satisfaction/requirements.txt:7]: done

File: customer-churn/kubeflow_requirements.txt [hunk 5]
```diff
@@ -1,9 +1,8 @@
 feature_engine==1.2.0
-imbalanced_learn==0.9.0
-imblearn==0.0
 lightgbm==3.3.2
-statsmodels
 xgboost
-zenml==0.7.3
+zenml==0.8.1rc0
```
[the remark this answers is not in the document]
Comment (stefannica) [reply customer-churn/kubeflow_requirements.txt]: done

File: customer-churn/requirements.txt [hunk 6]
```diff
@@ -1,8 +1,9 @@
...
-imblearn==0.0
-lightgbm==3.3.2
-statsmodels
+# lightgbm==3.3.2
+# statsmodels
 xgboost
-zenml==0.7.3
-optuna 
\ No newline at end of file
+zenml==0.8.1rc0
```
[the remark this answers is not in the document]
Comment (stefannica) [reply customer-churn/requirements.txt]: done

File: customer-churn/kubeflow_requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (stefannica) [reply customer-churn/kubeflow_requirements.txt]: removed mlflow and all integrations from requirements.txt and added them as pipeline required integrations in the code

File: customer-churn/requirements.txt [hunk 7]
```diff
@@ -1,8 +1,9 @@
...
+optuna 
+sklearn
+facets_overview
+mlflow
```
[the remark this answers is not in the document]
Comment (stefannica) [reply customer-churn/requirements.txt]: done

File: customer-churn/kubeflow_requirements.txt (hunk 1 above)
Comment (stefannica) [reply customer-churn/kubeflow_requirements.txt]: removed this file

File: customer-churn/run_seldon_deployment_pipeline.py (hunk 3 above)
Comment (stefannica) [reply customer-churn/run_seldon_deployment_pipeline.py]: done

**Comment by stefannica** [comment]
I updated the readme file and the requirements are now centralized in the requirements.txt file. I also included the last changes to how we manage stacks with `zenml stack up` and seldon core secrets.

**Review by stefannica — APPROVED** [review]
I ran a complete test by following/updating the readme file for customer-churn and everything worked out as expected.

**Pull request merged**